Traceback (most recent call last):
  File "C:\Users\Lan\anaconda3\envs\gds2023\lib\site-packages\jupyter_cache\executors\utils.py", line 58, in single_nb_execution
    executenb(
  File "C:\Users\Lan\anaconda3\envs\gds2023\lib\site-packages\nbclient\client.py", line 1204, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
  File "C:\Users\Lan\anaconda3\envs\gds2023\lib\site-packages\nbclient\util.py", line 84, in wrapped
    return just_run(coro(*args, **kwargs))
  File "C:\Users\Lan\anaconda3\envs\gds2023\lib\site-packages\nbclient\util.py", line 62, in just_run
    return loop.run_until_complete(coro)
  File "C:\Users\Lan\anaconda3\envs\gds2023\lib\asyncio\base_events.py", line 649, in run_until_complete
    return future.result()
  File "C:\Users\Lan\anaconda3\envs\gds2023\lib\site-packages\nbclient\client.py", line 663, in async_execute
    await self.async_execute_cell(
  File "C:\Users\Lan\anaconda3\envs\gds2023\lib\site-packages\nbclient\client.py", line 965, in async_execute_cell
    await self._check_raise_for_error(cell, cell_index, exec_reply)
  File "C:\Users\Lan\anaconda3\envs\gds2023\lib\site-packages\nbclient\client.py", line 862, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
# Convex Hull of our MultiPoint
convex = multi_point.convex_hull

# How many lines do we have inside our MultiLineString?
lines_count = len(multi_line)

# Let's calculate the area of our MultiPolygon
multi_poly_area = multi_poly.area

# 可以访问几何集合中的不同项。
west_area = multi_poly[0].area

# 可以检测几何对象是否有效，多边形的环不能有重叠，线几何不能自相交
valid = multi_poly.is_valid

print("Convex hull of the points: ", convex)
print("Number of lines in MultiLineString:", lines_count)
print("Area of our MultiPolygon:", multi_poly_area)
print("Area of our Western Hemisphere polygon:", west_area)
print("Is polygon valid?: ", valid)
------------------

[1;31m---------------------------------------------------------------------------[0m
[1;31mTypeError[0m                                 Traceback (most recent call last)
Cell [1;32mIn[32], line 5[0m
[0;32m      2[0m convex [38;5;241m=[39m multi_point[38;5;241m.[39mconvex_hull
[0;32m      4[0m [38;5;66;03m# How many lines do we have inside our MultiLineString?[39;00m
[1;32m----> 5[0m lines_count [38;5;241m=[39m [38;5;28;43mlen[39;49m[43m([49m[43mmulti_line[49m[43m)[49m
[0;32m      7[0m [38;5;66;03m# Let's calculate the area of our MultiPolygon[39;00m
[0;32m      8[0m multi_poly_area [38;5;241m=[39m multi_poly[38;5;241m.[39marea

[1;31mTypeError[0m: object of type 'MultiLineString' has no len()
TypeError: object of type 'MultiLineString' has no len()

